packages/core/src/lib/connection_manager.ts

```diff
@@ -456,6 +456,17 @@ export class ConnectionManager
       return false;
     }
 
+    const peer = await this.libp2p.peerStore.get(peerId);
+
+    if (
+      !peer.addresses.some((val) => val.multiaddr.toString().includes("/wss/"))
```

Comment:
know it's not for review yet but came here to mention the address can also be of the format `/tls/ws/`